app/javascript/App/bundles/ExchangeListings/client.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component, Fragment } from 'react'
+import debounce from 'debounce'
+import LayoutDesktop from '../../components/LayoutDesktop'
+import LayoutTablet from '../../components/LayoutTablet'
+import LayoutMobile from '../../components/LayoutMobile'
+import ListingsHeader from './components/ListingsHeader'
+import ListingsList from './components/ListingsList'
+import BodySection from './components/BodySection'
+
+class ExchangeListingsPage extends Component {
+  state = { showFilterPanel: false }
+
+  toggleFilterPanel() {
+    this.setState({
+      showFilterPanel: !this.state.showFilterPanel,
+    })
+  }
+
+  componentDidUpdate() {
+    window.addEventListener('resize', debounce(() => this.forceUpdate(), 500))
+  }
+
+  render() {
+    const props = this.props
+
+    if (window.isMobile) {
+      console.log('mobile')
+      return (
+        <LayoutMobile
+          {...props}
+          mainSection={
+            <Fragment>
+              <ListingsHeader
+                {...props}
+                showFilterPanel={this.state.showFilterPanel}
+                toggleFilterPanel={this.toggleFilterPanel.bind(this)}
```

Comment:
Instead of using `bind` (which creates a new function on every re-render) you can use instance properties:

```js
// `this` is auto-bound
toggleFilterPanel = () => ...;

toggleFilterPanel={this.toggleFilterPanel}
```